<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Run the quanTIseq algorithm — run_quantiseq • quantiseqr</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Run the quanTIseq algorithm — run_quantiseq" />
<meta property="og:description" content="Use quanTIseq to deconvolute a gene expression matrix." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



<!-- dependencies from examples -->


  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">quantiseqr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/using_quantiseqr.html">Using quantiseqr
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run the quanTIseq algorithm</h1>
    
    <div class="hidden name"><code>run_quantiseq.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Use quanTIseq to deconvolute a gene expression matrix.</p>
    </div>

    <pre class="usage"><span class='fu'>run_quantiseq</span><span class='op'>(</span>
  <span class='va'>expression_data</span>,
  signature_matrix <span class='op'>=</span> <span class='st'>"TIL10"</span>,
  is_arraydata <span class='op'>=</span> <span class='cn'>FALSE</span>,
  is_tumordata <span class='op'>=</span> <span class='cn'>FALSE</span>,
  scale_mRNA <span class='op'>=</span> <span class='cn'>TRUE</span>,
  method <span class='op'>=</span> <span class='st'>"lsei"</span>,
  column <span class='op'>=</span> <span class='st'>"gene_symbol"</span>,
  rm_genes <span class='op'>=</span> <span class='cn'>NULL</span>,
  return_se <span class='op'>=</span> <span class='fu'>is</span><span class='op'>(</span><span class='va'>expression_data</span>, <span class='st'>"SummarizedExperiment"</span><span class='op'>)</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>expression_data</th>
      <td><p>The gene expression information, containing the TPM
values for the measured features.
Can be provided as</p><ul>
<li><p>a simple gene expression matrix, or a data frame (with HGNC gene symbols as
row names and sample identifiers as column names)</p></li>
<li><p>an <code>ExpressionSet</code> object (from the Biobase package), where the HGNC gene symbols
are provided in a column of the <code>fData</code> slot - that is specified by the <code>column</code>
parameter below</p></li>
<li><p>a <code>SummarizedExperiment</code> object, or any of the derivative classes (e.g. DESeq2's
<code>DESeqDataSet</code>), in which the assay (default: "abundance") is containing the TPMs
as expected. Internally, <code>quantiseqr</code> handles the conversion to an object which
is used in the deconvolution procedure.</p></li>
</ul></td>
    </tr>
    <tr>
      <th>signature_matrix</th>
      <td><p>Character string, specifying the name of the signature matrix.
At the moment, only the original <code>TIL10</code> signature can be selected.</p></td>
    </tr>
    <tr>
      <th>is_arraydata</th>
      <td><p>Logical value. Should be set to TRUE if the expression data
are originating from microarray data. For RNA-seq data, this has to be FALSE
(default value). If set to TRUE, the <code>rmgenes</code> parameter (see below) is set
to "none".</p></td>
    </tr>
    <tr>
      <th>is_tumordata</th>
      <td><p>Logical value. Should be set to TRUE if the expression data
is from tumor samples. Default: FALSE (e.g. for RNA-seq from blood samples)</p></td>
    </tr>
    <tr>
      <th>scale_mRNA</th>
      <td><p>Logical value. If set to FALSE, it disables the correction
of cell-type-specific mRNA content bias. Default: TRUE</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>Character string, defining the deconvolution method to be used:
<code>lsei</code> for constrained least squares regression, <code>hampel</code>, <code>huber</code>, or <code>bisquare</code>
for robust regression with Huber, Hampel, or Tukey bisquare estimators,
respectively. Default: <code>lsei</code>.</p></td>
    </tr>
    <tr>
      <th>column</th>
      <td><p>Character, specifies which column in the <code>fData</code> slot (for the
ExpressionSet object) contains the information of the HGNC gene symbol
identifiers</p></td>
    </tr>
    <tr>
      <th>rm_genes</th>
      <td><p>Character vector, specifying which genes have to be excluded
from the deconvolution analysis. It can be provided as</p><ul>
<li><p>a vector of gene symbols (contained in the <code>expression_data</code>)</p></li>
<li><p>a single string among the choices of "none" (no genes are removed) and "default"
(a list of genes with noisy expression RNA-seq data is removed, as explained
in the quanTIseq paper).
Default: "default" for RNA-seq data, "none" for microarrays.</p></li>
</ul></td>
    </tr>
    <tr>
      <th>return_se</th>
      <td><p>Logical value, controls the format of how the quantification
is returned. If providing a <code>SummarizedExperiment</code>, it can simply extend its
<code>colData</code> component, without the need to create a separate data frame as output.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A data.frame containing the quantifications of the cell type proportions,
or alternatively, if providing <code>expression_data</code> as <code>SummarizedExperiment</code> and
setting <code>return_se</code> to TRUE, a <code>SummarizedExperiment</code> with the quantifications
included by expanding the <code>colData</code> slot of the original object</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The values contained in the <code>expression_data</code> need to be provided as
TPM values, as this is the format also used to store the <code>TIL10</code> signature, upon
which quanTIseq builds to perform the immune cell type deconvolution.
Expression data should <em>not</em> be provided in logarithmic scale.</p>
<p>If providing the <code>expression_data</code> as a <code>SummarizedExperiment</code>/<code>DESeqDataSet</code>
object, it might be beneficial that this has been created via <code>tximport</code> -
if this is the case, the assay named "abundance" will be automatically
created upon importing the transcript quantification results.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>F. Finotello, C. Mayer, C. Plattner, G. Laschober, D. Rieder,
H. Hackl, A. Krogsdam, Z. Loncova, W. Posch, D. Wilflingseder, S. Sopper,
M. Jsselsteijn, T. P. Brouwer, D. Johnsons, Y. Xu, Y. Wang, M. E. Sanders,
M. V. Estrada, P. Ericsson-Gonzalez, P. Charoentong, J. Balko,
N. F. d. C. C. de Miranda, Z. Trajanoski.
"Molecular and pharmacological modulators of the tumor immune contexture
revealed by deconvolution of RNA-seq data".
Genome Medicine 2019;11(1):34. doi: 10.1186/s13073-019-0638-6.</p>
<p>C. Plattner, F. Finotello, D. Rieder.
"Chapter Ten - Deconvoluting tumor-infiltrating immune cells from RNA-seq
data using quanTIseq".
Methods in Enzymology, 2020. doi: 10.1016/bs.mie.2019.05.056.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>dataset_racle</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>dataset_racle</span><span class='op'>$</span><span class='va'>expr_mat</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] 32467     4</div><div class='input'><span class='va'>res_quantiseq_run</span> <span class='op'>&lt;-</span> <span class='fu'>quantiseqr</span><span class='fu'>::</span><span class='fu'>run_quantiseq</span><span class='op'>(</span>
  expression_data <span class='op'>=</span> <span class='va'>dataset_racle</span><span class='op'>$</span><span class='va'>expr_mat</span>,
  signature_matrix <span class='op'>=</span> <span class='st'>"TIL10"</span>,
  is_arraydata <span class='op'>=</span> <span class='cn'>FALSE</span>,
  is_tumordata <span class='op'>=</span> <span class='cn'>TRUE</span>,
  scale_mRNA <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Running quanTIseq deconvolution module</span></div><div class='output co'>#&gt; <span class='message'>Gene expression normalization and re-annotation (arrays: FALSE)</span></div><div class='output co'>#&gt; <span class='message'>Removing 17 noisy genes</span></div><div class='output co'>#&gt; <span class='message'>Removing 15 genes with high expression in tumors</span></div><div class='output co'>#&gt; <span class='message'>Signature genes found in data set: 135/138 (97.83%)</span></div><div class='output co'>#&gt; <span class='message'>Mixture deconvolution (method: lsei)</span></div><div class='output co'>#&gt; <span class='message'>Deconvolution successful!</span></div><div class='input'>
<span class='co'># using a SummarizedExperiment object</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://bioconductor.org/packages/SummarizedExperiment'>"SummarizedExperiment"</a></span><span class='op'>)</span>
<span class='va'>se_racle</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>SummarizedExperiment</a></span><span class='op'>(</span>
  assays <span class='op'>=</span> <span class='fu'>List</span><span class='op'>(</span>
    abundance <span class='op'>=</span> <span class='va'>dataset_racle</span><span class='op'>$</span><span class='va'>expr_mat</span>
  <span class='op'>)</span>,
  colData <span class='op'>=</span> <span class='fu'>DataFrame</span><span class='op'>(</span>
    SampleName <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>dataset_racle</span><span class='op'>$</span><span class='va'>expr_mat</span><span class='op'>)</span>
  <span class='op'>)</span>
<span class='op'>)</span>

<span class='va'>res_run_SE</span> <span class='op'>&lt;-</span> <span class='fu'>quantiseqr</span><span class='fu'>::</span><span class='fu'>run_quantiseq</span><span class='op'>(</span>
    expression_data <span class='op'>=</span> <span class='va'>se_racle</span>,
    signature_matrix <span class='op'>=</span> <span class='st'>"TIL10"</span>,
    is_arraydata <span class='op'>=</span> <span class='cn'>FALSE</span>,
    is_tumordata <span class='op'>=</span> <span class='cn'>TRUE</span>,
    scale_mRNA <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Running quanTIseq deconvolution module</span></div><div class='output co'>#&gt; <span class='message'>Gene expression normalization and re-annotation (arrays: FALSE)</span></div><div class='output co'>#&gt; <span class='message'>Removing 17 noisy genes</span></div><div class='output co'>#&gt; <span class='message'>Removing 15 genes with high expression in tumors</span></div><div class='output co'>#&gt; <span class='message'>Signature genes found in data set: 135/138 (97.83%)</span></div><div class='output co'>#&gt; <span class='message'>Mixture deconvolution (method: lsei)</span></div><div class='output co'>#&gt; <span class='message'>Deconvolution successful!</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Federico Marini, Francesca Finotello.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


