<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using quantiseqr
 • quantiseqr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using quantiseqr
">
<meta property="og:description" content="quantiseqr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">quantiseqr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/using_quantiseqr.html">Using quantiseqr
</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="using_quantiseqr_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
<p>Using quantiseqr</p>
                        <h4 data-toc-skip class="author">Federico Marini<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics (<a href="https://www.unimedizin-mainz.de/imbei/imbei/welcome-page.html?L=1" class="external-link">IMBEI, Mainz</a>)<br><h4 data-toc-skip class="author">Francesca Finotello<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>
</h4>
            <address class="author_afil">
      Institute of Bioinformatics, Biocenter Medical University of Innsbruck (<a href="https://icbi.i-med.ac.at/index.html" class="external-link uri">https://icbi.i-med.ac.at/index.html</a>)<br><h4 data-toc-skip class="date">2021-04-24</h4>
      
      
      <div class="hidden name"><code>using_quantiseqr.Rmd</code></div>

    </address>
</address>
</div>

    
    
<style type="text/css">
.smaller {
  font-size: 10px
}
</style>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor" aria-hidden="true"></a>Introduction</h1>
<p>This vignette describes how to use the <em><a href="https://bioconductor.org/packages/3.13/quantiseqr" class="external-link">quantiseqr</a></em> package for streamlining your workflow around the quanTIseq method. This was developed to perform the quantification of the Tumor Immune contexture from RNA-seq data. The TIL10 signature, designed in this context, is used to dissect the contributions of ten immune cell types, and has bee carefully crafted from a collection of human RNA-seq samples. The TIL10 signature has been extensively validated using simulated, flow cytometry, and immunohistochemistry data.</p>
</div>
<div id="gettingstarted" class="section level1">
<h1 class="hasAnchor">
<a href="#gettingstarted" class="anchor" aria-hidden="true"></a>Getting started</h1>
<p>To install this package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"quantiseqr"</span><span class="op">)</span></code></pre></div>
<p>Once installed, the package can be loaded and attached to your current workspace as follows:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"quantiseqr"</span><span class="op">)</span></code></pre></div>
<p>In the following chunk, we load a set of additional packages that will be required throughout this vignette.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://tidyr.tidyverse.org" class="external-link">"tidyr"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://tibble.tidyverse.org/" class="external-link">"tibble"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/seandavi/GEOquery" class="external-link">"GEOquery"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/hadley/reshape" class="external-link">"reshape2"</a></span><span class="op">)</span></code></pre></div>
<p>In order to use <em><a href="https://bioconductor.org/packages/3.13/quantiseqr" class="external-link">quantiseqr</a></em> in your workflow, one fundamental input is required, to be provided to <code>run_quantiseqr()</code> as <code>expression_data</code>. This is an object containing the TPM expression values for the measured features, and can be provided in different ways:</p>
<ul>
<li>as a simple gene expression matrix, or a data frame (with HGNC gene symbols as row names and sample identifiers as column names)</li>
<li>as an <code>ExpressionSet</code> object (from the <em><a href="https://bioconductor.org/packages/3.13/Biobase" class="external-link">Biobase</a></em> package), where the HGNC gene symbols are provided in a column of the <code>fData</code> slot</li>
<li>as a <code>SummarizedExperiment</code> object, or any of the derivative classes (e.g. <em><a href="https://bioconductor.org/packages/3.13/DESeq2" class="external-link">DESeq2</a></em>’s <code>DESeqDataSet</code>), in which the assay (default: “abundance”) is containing the TPMs as expected</li>
</ul>
</div>
<div id="some-use-cases-for-quantiseqr" class="section level1">
<h1 class="hasAnchor">
<a href="#some-use-cases-for-quantiseqr" class="anchor" aria-hidden="true"></a>Some use cases for <code>quantiseqr</code>
</h1>
<p>In this section, we illustrate the usage of <code>quantiseqr</code> on a variety of datasets. These differ with respect to their size and samples of origin, and we illustrate how the different parameters of <code>quantiseqr</code> should be set in the different scenarios.</p>
<p>The fundamental input for <code>quantiseqr</code> is a gene expression matrix-like object, with features on the rows, and samples as the columns. <code>quantiseqr</code> can also directly handle <code>SummarizedExperiment</code> objects, as well as <code>ExpressionSet</code> objects, commonly used for microarray data. In case a <code>SummarizedExperiment</code> object is passed, the quantifications of the immune contexture can be directly returned extending the <code>colData</code> of the provided input.</p>
<div id="use-case-1-metastatic-melanoma-patients-racle-et-al-2017" class="section level2">
<h2 class="hasAnchor">
<a href="#use-case-1-metastatic-melanoma-patients-racle-et-al-2017" class="anchor" aria-hidden="true"></a>Use case 1: Metastatic melanoma patients (Racle et al 2017)</h2>
<p><code>quantiseqr</code> ships with an example dataset with samples from four patients with metastatic melanoma published in <span class="citation">(Racle et al. 2017)</span>.</p>
<p>The dataset <code><a href="../reference/dataset_racle.html">quantiseqr::dataset_racle</a></code> contains - a gene expression matrix (<code>dataset_racle$expr_mat</code>) generated using bulk RNA-seq - ‘gold standard’ estimates of immune cell contents profiled with FACS (<code>dataset_racle$ref</code>).</p>
<p>We are going to use the bulk RNA-seq data to run the deconvolution methods and will compare the results to the FACS data in the following steps.</p>
<p>Let’s inspect the expression matrix first:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"dataset_racle"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">dataset_racle</span><span class="op">$</span><span class="va">expr_mat</span><span class="op">)</span>
<span class="co">#&gt; [1] 32467     4</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">dataset_racle</span><span class="op">$</span><span class="va">expr_mat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">LAU125</th>
<th align="right">LAU355</th>
<th align="right">LAU1255</th>
<th align="right">LAU1314</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">A1BG</td>
<td align="right">0.82</td>
<td align="right">0.58</td>
<td align="right">0.81</td>
<td align="right">0.71</td>
</tr>
<tr class="even">
<td align="left">A1CF</td>
<td align="right">0.00</td>
<td align="right">0.01</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td align="left">A2M</td>
<td align="right">247.15</td>
<td align="right">24.88</td>
<td align="right">2307.94</td>
<td align="right">20.30</td>
</tr>
<tr class="even">
<td align="left">A2M-AS1</td>
<td align="right">1.38</td>
<td align="right">0.20</td>
<td align="right">2.60</td>
<td align="right">0.28</td>
</tr>
<tr class="odd">
<td align="left">A2ML1</td>
<td align="right">0.03</td>
<td align="right">0.00</td>
<td align="right">0.05</td>
<td align="right">0.02</td>
</tr>
</tbody>
</table>
<p>The quantification of the immune cell types with <code>quantiseqr</code> can be done as in the chunk below:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ti_racle</span> <span class="op">&lt;-</span> <span class="fu">quantiseqr</span><span class="fu">::</span><span class="fu"><a href="../reference/run_quantiseq.html">run_quantiseq</a></span><span class="op">(</span>
  expression_data <span class="op">=</span> <span class="va">dataset_racle</span><span class="op">$</span><span class="va">expr_mat</span>,
  signature_matrix <span class="op">=</span> <span class="st">"TIL10"</span>,
  is_arraydata <span class="op">=</span> <span class="cn">FALSE</span>,
  is_tumordata <span class="op">=</span> <span class="cn">TRUE</span>,
  scale_mRNA <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running quanTIseq deconvolution module</span>
<span class="co">#&gt; Gene expression normalization and re-annotation (arrays: FALSE)</span>
<span class="co">#&gt; Removing 17 noisy genes</span>
<span class="co">#&gt; Removing 15 genes with high expression in tumors</span>
<span class="co">#&gt; Signature genes found in data set: 135/138 (97.83%)</span>
<span class="co">#&gt; Mixture deconvolution (method: lsei)</span>
<span class="co">#&gt; Deconvolution successful!</span></code></pre></div>
<p>The call above means that we are passing the expression data as simple matrix (in <code>dataset_racle$expr_mat</code>) and quantifying the tumor immune contexture using the (default) TIL10 signature. This is a dataset stemming from tumor RNA-seq samples (therefore the values specified for <code>is_tumordata</code> and <code>is_arraydata</code>). With <code>scale_mRNA</code> set to <code>TRUE</code>, we are performing the correction of cell-type-specific mRNA content bias.</p>
<p>The output of <code>quantiseqr</code> can be further processed and represented in a tabular or visual manner to facilitate the comparisons across samples/conditions.</p>
<p>The scores returned by <code>quantiseqr</code> can be interpreted as a cell-type fraction, making it possible to represent them as a stacked bar chart.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/quantiplot.html">quantiplot</a></span><span class="op">(</span><span class="va">ti_racle</span><span class="op">)</span></code></pre></div>
<p><img src="using_quantiseqr_files/figure-html/ex1-ti-plot-1.png" width="768"></p>
<p>We observe that</p>
<ul>
<li>two samples (LAU355, LAU1314) appear to contain a large amount of CD4+ T cells and B cells</li>
<li>the other two samples (LAU1255, LAU125) appear to contain a large amount of “uncharacterized cells”</li>
<li>one sample (LAU125) appears to contain no CD8+ T cells, often associated with bad prognosis.</li>
</ul>
<p>Estimating the amount of “uncharacterized cells” is a novel feature introduced by quanTIseq and EPIC <span class="citation">(Finotello et al. 2019, @EPIC2017)</span>. This estimate often corresponds to the fraction of tumor cells in the sample.</p>
<!-- compare with facs internal refs? -->
</div>
<div id="use-case-2-pbmcs-from-gse107572-finotello-et-al-2019" class="section level2">
<h2 class="hasAnchor">
<a href="#use-case-2-pbmcs-from-gse107572-finotello-et-al-2019" class="anchor" aria-hidden="true"></a>Use case 2: PBMCs from GSE107572 (Finotello et al 2019)</h2>
<p>Here we show how to use <code>quantiseqr</code> to deconvolute blood-derived immune-cell mixtures <span class="citation">(Finotello et al. 2019)</span>, for which also matching flow cytometry data are available.</p>
<p>This is also presented as an example in <span class="citation">(Plattner, Finotello, and Rieder 2020)</span>, please refer to this later publication for additional details on the processing steps.</p>
<p>The example dataset is available online at the Gene Expression Omnibus (accession number GSE107572), and is provided as preprocessed RNA-seq data from blood-derived immune-cell mixtures from nine healthy donors.<br>
Flow cytometry estimates for the according immune subpopulations are also available</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## While downloading by hand is possible, it is recommended to use GEOquery</span>
<span class="co"># wget -c ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE107nnn/GSE107572/suppl/GSE107572%5Ftpm%5FPBMC%5FRNAseq%2Etxt%2Egz</span>
<span class="co"># unzip GSE107572_tpm_PBMC_RNAseq.txt.gz</span>
<span class="co"># read.table("GSE107572_tpm_PBMC_RNAseq.txt", header = TRUE)</span>


<span class="co"># downloading the supplemental files on the fly</span>
<span class="va">tpminfo_GSE107572</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GEOquery/man/getGEOSuppFiles.html" class="external-link">getGEOSuppFiles</a></span><span class="op">(</span><span class="st">"GSE107572"</span>,
  baseDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,
  filter_regex <span class="op">=</span> <span class="st">"GSE107572_tpm_PBMC_RNAseq"</span>
<span class="op">)</span>
<span class="va">tpm_location</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tpminfo_GSE107572</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="va">tpm_location</span>
<span class="co">#&gt; [1] "/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//Rtmp768T85/GSE107572/GSE107572_tpm_PBMC_RNAseq.txt.gz"</span>
<span class="va">tpmdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">tpm_location</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tpm_genesymbols</span> <span class="op">&lt;-</span> <span class="va">tpmdata</span><span class="op">$</span><span class="va">GENE</span>
<span class="va">tpmdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">tpmdata</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tpmdata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tpm_genesymbols</span>

<span class="va">ti_PBMCs</span> <span class="op">&lt;-</span> <span class="fu">quantiseqr</span><span class="fu">::</span><span class="fu"><a href="../reference/run_quantiseq.html">run_quantiseq</a></span><span class="op">(</span>
  expression_data <span class="op">=</span> <span class="va">tpmdata</span>,
  signature_matrix <span class="op">=</span> <span class="st">"TIL10"</span>,
  is_arraydata <span class="op">=</span> <span class="cn">FALSE</span>,
  is_tumordata <span class="op">=</span> <span class="cn">FALSE</span>,
  scale_mRNA <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running quanTIseq deconvolution module</span>
<span class="co">#&gt; Gene expression normalization and re-annotation (arrays: FALSE)</span>
<span class="co">#&gt; Removing 17 noisy genes</span>
<span class="co">#&gt; Signature genes found in data set: 149/153 (97.39%)</span>
<span class="co">#&gt; Mixture deconvolution (method: lsei)</span>
<span class="co">#&gt; Deconvolution successful!</span></code></pre></div>
<p>To obtain an overview on the cell type compositions, we can print out the result and plot that with the conveniente <code>quantiplot</code> wrapper:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ti_PBMCs</span>
<span class="co">#&gt;                                        Sample    B.cells Macrophages.M1</span>
<span class="co">#&gt; pbmc_1                                 pbmc_1 0.09386998              0</span>
<span class="co">#&gt; pbmc_10                               pbmc_10 0.09625346              0</span>
<span class="co">#&gt; pbmc_12_merged_1.fastq pbmc_12_merged_1.fastq 0.06564512              0</span>
<span class="co">#&gt; pbmc_2_merged_1.fastq   pbmc_2_merged_1.fastq 0.04788587              0</span>
<span class="co">#&gt; pbmc_4                                 pbmc_4 0.07113262              0</span>
<span class="co">#&gt; pbmc_5_merged_1.fastq   pbmc_5_merged_1.fastq 0.08526253              0</span>
<span class="co">#&gt; pbmc_6_merged_1.fastq   pbmc_6_merged_1.fastq 0.08695852              0</span>
<span class="co">#&gt; pbmc_7_merged_1.fastq   pbmc_7_merged_1.fastq 0.04749548              0</span>
<span class="co">#&gt; pbmc_9_merged_1.fastq   pbmc_9_merged_1.fastq 0.04637309              0</span>
<span class="co">#&gt;                        Macrophages.M2 Monocytes Neutrophils    NK.cells</span>
<span class="co">#&gt; pbmc_1                    0.017214191 0.1709194 0.022854647 0.023960144</span>
<span class="co">#&gt; pbmc_10                   0.009658902 0.2064216 0.026710342 0.004535281</span>
<span class="co">#&gt; pbmc_12_merged_1.fastq    0.010348082 0.1070846 0.014807965 0.061400801</span>
<span class="co">#&gt; pbmc_2_merged_1.fastq     0.016706096 0.1500564 0.050937018 0.026005414</span>
<span class="co">#&gt; pbmc_4                    0.002411796 0.1339592 0.009949874 0.102721684</span>
<span class="co">#&gt; pbmc_5_merged_1.fastq     0.007025086 0.2333513 0.011131749 0.082349782</span>
<span class="co">#&gt; pbmc_6_merged_1.fastq     0.018465121 0.2116348 0.050735601 0.021043316</span>
<span class="co">#&gt; pbmc_7_merged_1.fastq     0.019879914 0.1669343 0.034329149 0.079715496</span>
<span class="co">#&gt; pbmc_9_merged_1.fastq     0.017734011 0.1910473 0.039968231 0.059074869</span>
<span class="co">#&gt;                        T.cells.CD4 T.cells.CD8       Tregs Dendritic.cells</span>
<span class="co">#&gt; pbmc_1                  0.29483429   0.2254946 0.038462244      0.02140064</span>
<span class="co">#&gt; pbmc_10                 0.23050466   0.1973788 0.033000355      0.03900199</span>
<span class="co">#&gt; pbmc_12_merged_1.fastq  0.10126385   0.3668420 0.009534643      0.07088375</span>
<span class="co">#&gt; pbmc_2_merged_1.fastq   0.23184199   0.1894586 0.028859312      0.02085107</span>
<span class="co">#&gt; pbmc_4                  0.11342624   0.3417581 0.042472502      0.07729367</span>
<span class="co">#&gt; pbmc_5_merged_1.fastq   0.05931314   0.4153910 0.026181427      0.05987356</span>
<span class="co">#&gt; pbmc_6_merged_1.fastq   0.13491730   0.3102470 0.018981104      0.02072611</span>
<span class="co">#&gt; pbmc_7_merged_1.fastq   0.00000000   0.4843902 0.000000000      0.03288725</span>
<span class="co">#&gt; pbmc_9_merged_1.fastq   0.07959930   0.2757124 0.010779655      0.05363633</span>
<span class="co">#&gt;                             Other</span>
<span class="co">#&gt; pbmc_1                 0.09098993</span>
<span class="co">#&gt; pbmc_10                0.15653461</span>
<span class="co">#&gt; pbmc_12_merged_1.fastq 0.19218912</span>
<span class="co">#&gt; pbmc_2_merged_1.fastq  0.23739827</span>
<span class="co">#&gt; pbmc_4                 0.10487431</span>
<span class="co">#&gt; pbmc_5_merged_1.fastq  0.02012044</span>
<span class="co">#&gt; pbmc_6_merged_1.fastq  0.12629107</span>
<span class="co">#&gt; pbmc_7_merged_1.fastq  0.13436824</span>
<span class="co">#&gt; pbmc_9_merged_1.fastq  0.22607486</span>
<span class="fu"><a href="../reference/quantiplot.html">quantiplot</a></span><span class="op">(</span><span class="va">ti_PBMCs</span><span class="op">)</span></code></pre></div>
<p><img src="using_quantiseqr_files/figure-html/ex2-ti-plot-1.png" width="700"></p>
<p>Notably, for these samples, corresponding quantifications of the true cell fractions done by flow cytometry are also available.<br>
In the chunk that follows, we retrieve that information and generate scatter plots to display how the estimated values from <code>quantiseqr</code> correlate with the ground truth values.<br>
This is adapted by <span class="citation">(Plattner, Finotello, and Rieder 2020)</span> to match the output format generated by <code>quantiseqr</code> - and can be used as a template in case other types of matched ground-truth information are available.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">GEOid</span> <span class="op">&lt;-</span> <span class="st">"GSE107572"</span>
<span class="va">gds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GEOquery/man/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="va">GEOid</span><span class="op">)</span>
<span class="co">#&gt; Found 1 file(s)</span>
<span class="co">#&gt; GSE107572_series_matrix.txt.gz</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ── Column specification ────────────────────────────────────────────────────────</span>
<span class="co">#&gt; cols(</span>
<span class="co">#&gt;   ID_REF = col_character(),</span>
<span class="co">#&gt;   GSM2871599 = col_character(),</span>
<span class="co">#&gt;   GSM2871600 = col_character(),</span>
<span class="co">#&gt;   GSM2871601 = col_character(),</span>
<span class="co">#&gt;   GSM2871602 = col_character(),</span>
<span class="co">#&gt;   GSM2871603 = col_character(),</span>
<span class="co">#&gt;   GSM2871604 = col_character(),</span>
<span class="co">#&gt;   GSM2871605 = col_character(),</span>
<span class="co">#&gt;   GSM2871606 = col_character(),</span>
<span class="co">#&gt;   GSM2871607 = col_character()</span>
<span class="co">#&gt; )</span>
<span class="co">#&gt; File stored at:</span>
<span class="co">#&gt; /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T//Rtmp768T85/GPL16791.soft</span>
<span class="va">GEOinfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gds</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="va">FACSdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  B.cells <span class="op">=</span> <span class="va">GEOinfo</span><span class="op">$</span><span class="va">`b cells:ch1`</span>,
  T.cells.CD4 <span class="op">=</span> <span class="va">GEOinfo</span><span class="op">$</span><span class="va">`cd4+ t cells:ch1`</span>,
  T.cells.CD8 <span class="op">=</span> <span class="va">GEOinfo</span><span class="op">$</span><span class="va">`cd8+ t cells:ch1`</span>,
  Monocytes <span class="op">=</span> <span class="va">GEOinfo</span><span class="op">$</span><span class="va">`monocytes:ch1`</span>,
  Dendritic.cells <span class="op">=</span> <span class="va">GEOinfo</span><span class="op">$</span><span class="va">`myeloid dendritic cells:ch1`</span>,
  NK.cells <span class="op">=</span> <span class="va">GEOinfo</span><span class="op">$</span><span class="va">`natural killer cells:ch1`</span>,
  Neutrophils <span class="op">=</span> <span class="va">GEOinfo</span><span class="op">$</span><span class="va">`neutrophils:ch1`</span>,
  Tregs <span class="op">=</span> <span class="va">GEOinfo</span><span class="op">$</span><span class="va">`tregs:ch1`</span>
<span class="op">)</span>
<span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">FACSdata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span>
  <span class="st">"Blood-derived immune-cell mixture from donor "</span>, <span class="st">"pbmc"</span>, <span class="va">GEOinfo</span><span class="op">$</span><span class="va">title</span>
<span class="op">)</span>

<span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ti_PBMCs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_.*$"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"_"</span>, <span class="st">""</span>, <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ti_PBMCs</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="va">ccells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ti_PBMCs</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">FACSdata</span><span class="op">)</span><span class="op">)</span>
<span class="va">csbjs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ti_PBMCs</span><span class="op">)</span>, <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">FACSdata</span><span class="op">)</span><span class="op">)</span>

<span class="va">ti_PBMCs</span> <span class="op">&lt;-</span> <span class="va">ti_PBMCs</span><span class="op">[</span><span class="va">csbjs</span>, <span class="va">ccells</span><span class="op">]</span>
<span class="va">FACSdata</span> <span class="op">&lt;-</span> <span class="va">FACSdata</span><span class="op">[</span><span class="va">csbjs</span>, <span class="va">ccells</span><span class="op">]</span>

<span class="va">palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#451C87"</span>, <span class="st">"#B3B300"</span>, <span class="st">"#CE0648"</span>, <span class="st">"#2363C5"</span>, <span class="st">"#AB4CA1"</span>, <span class="st">"#0A839B"</span>, <span class="st">"#DD8C24"</span>, <span class="st">"#ED6D42"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">palette</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"T.cells.CD4"</span>, <span class="st">"Dendritic.cells"</span>, <span class="st">"Monocytes"</span>, <span class="st">"T.cells.CD8"</span>, <span class="st">"Tregs"</span>, <span class="st">"B.cells"</span>, <span class="st">"NK.cells"</span>, <span class="st">"Neutrophils"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="va">colall</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">ti_PBMCs</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
  <span class="kw">if</span> <span class="op">(</span><span class="va">i</span> <span class="op">&lt;=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">ti_PBMCs</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">FACSdata</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
    <span class="va">y</span> <span class="op">&lt;-</span> <span class="va">ti_PBMCs</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span>
    <span class="va">ccell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">ti_PBMCs</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>
    <span class="va">col</span> <span class="op">&lt;-</span> <span class="va">palette</span><span class="op">[</span><span class="va">ccell</span><span class="op">]</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">FACSdata</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
    <span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">ti_PBMCs</span><span class="op">)</span><span class="op">)</span>
    <span class="va">ccell</span> <span class="op">&lt;-</span> <span class="st">"All cells"</span>
    <span class="va">col</span> <span class="op">&lt;-</span> <span class="va">colall</span>
  <span class="op">}</span>
  <span class="va">res.cor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span><span class="va">y</span>, <span class="va">x</span><span class="op">)</span>
  <span class="va">R</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">res.cor</span><span class="op">$</span><span class="va">estimate</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/format.pval.html" class="external-link">format.pval</a></span><span class="op">(</span><span class="va">res.cor</span><span class="op">$</span><span class="va">p.value</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
  <span class="va">RMSE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="op">(</span><span class="va">y</span> <span class="op">-</span> <span class="va">x</span><span class="op">)</span><span class="op">^</span><span class="fl">2</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>

  <span class="va">regl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">y</span> <span class="op">~</span> <span class="va">x</span><span class="op">)</span>
  <span class="va">ymax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1.3</span>, <span class="fl">0.01</span><span class="op">)</span>
  <span class="va">xmax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>, digits <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>, <span class="fl">0.01</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>,
    main <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(\\.)"</span>, <span class="st">" "</span>, <span class="va">ccell</span><span class="op">)</span>, pch <span class="op">=</span> <span class="fl">19</span>,
    xlab <span class="op">=</span> <span class="st">"Flow cytometry cell fractions"</span>,
    ylab <span class="op">=</span> <span class="st">"quanTIseq cell fractions"</span>,
    col <span class="op">=</span> <span class="va">col</span>, cex.main <span class="op">=</span> <span class="fl">1.3</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">ymax</span><span class="op">)</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">xmax</span><span class="op">)</span>, las <span class="op">=</span> <span class="fl">1</span>
  <span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="va">regl</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/text.html" class="external-link">text</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">ymax</span> <span class="op">*</span> <span class="fl">0.98</span>, cex <span class="op">=</span> <span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"r = "</span>, <span class="va">R</span>, <span class="st">", p = "</span>, <span class="va">p</span><span class="op">)</span>, pos <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/graphics/text.html" class="external-link">text</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">ymax</span> <span class="op">*</span> <span class="fl">0.9</span>, cex <span class="op">=</span> <span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"RMSE = "</span>, <span class="va">RMSE</span><span class="op">)</span>, pos <span class="op">=</span> <span class="fl">4</span><span class="op">)</span>

  <span class="va">colall</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">colall</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">col</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<p><img src="using_quantiseqr_files/figure-html/ex2-comparison-1.png" width="864"></p>
</div>
<div id="use-case-3-expression-changes-in-melanomas-pre-mapki-treatment-vs--on-mapki-treatment---gse75299-song-et-al-2017" class="section level2">
<h2 class="hasAnchor">
<a href="#use-case-3-expression-changes-in-melanomas-pre-mapki-treatment-vs--on-mapki-treatment---gse75299-song-et-al-2017" class="anchor" aria-hidden="true"></a>Use case 3: Expression changes in Melanomas pre MAPKi treatment vs. on MAPKi treatment - GSE75299 (Song et al 2017)</h2>
<p>We use here the dataset provided in <span class="citation">(Song et al. 2017)</span>, where patients and cell lines, before and on treatment with MAPK inhibitors are quantified with RNA-seq.</p>
<p>The original dataset is available via GEO at the accession GSE75299, but we will be loading a preprocessed version of it, containing the TPM expression values, made available via <code>ExperimentHub</code>.<br>
This will enable us to performed a paired analysis on the same observation units (the patients) to appreciate differences in the cell proportions induced by the MAPKi treatment.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_already_on_ehub</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### WAS...</span>
<span class="va">tpmdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tpm_GSE75299.txt.gz"</span>, package <span class="op">=</span> <span class="st">"quantiseqr"</span><span class="op">)</span>,
  header <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>

<span class="co">### WILL BE...</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"ExperimentHub"</span><span class="op">)</span>
<span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="op">(</span><span class="op">)</span>
<span class="va">quantiseqdata_eh</span> <span class="op">&lt;-</span> <span class="fu">query</span><span class="op">(</span><span class="va">eh</span>, <span class="st">"quantiseqrData"</span><span class="op">)</span>
<span class="va">se_Song2017_MAPKi_treatment</span> <span class="op">&lt;-</span> <span class="va">quantiseqdata_eh</span><span class="op">[[</span><span class="st">"EH123"</span><span class="op">]</span><span class="op">]</span> <span class="co"># will be edited</span>

<span class="co">#### tpm_genesymbols &lt;- tpmdata$GENE</span>
<span class="co">#### tpmdata &lt;- as.matrix(tpmdata[, -1])</span>
<span class="co">#### rownames(tpmdata) &lt;- tpm_genesymbols</span>

<span class="co">#### ti_quant &lt;- quantiseqr::run_quantiseq(</span>
<span class="co">####   expression_data = se_Song2017_MAPKi_treatment,</span>
<span class="co">####   signature_matrix = "TIL10",</span>
<span class="co">####   is_arraydata = FALSE,</span>
<span class="co">####   is_tumordata = TRUE,</span>
<span class="co">####   scale_mRNA = TRUE</span>
<span class="co">#### )</span>

<span class="va">Song2017_MAPKi_treatment_tiquant</span> <span class="op">&lt;-</span> <span class="fu">quantiseqr</span><span class="fu">::</span><span class="fu"><a href="../reference/run_quantiseq.html">run_quantiseq</a></span><span class="op">(</span>
  expression_data <span class="op">=</span> <span class="va">Song2017_MAPKi_treatment</span>,
  signature_matrix <span class="op">=</span> <span class="st">"TIL10"</span>,
  is_arraydata <span class="op">=</span> <span class="cn">FALSE</span>,
  is_tumordata <span class="op">=</span> <span class="cn">TRUE</span>,
  scale_mRNA <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>


<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Song2017_MAPKi_treatment_tiquant</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">Song2017_MAPKi_treatment_tiquant</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">Song2017_MAPKi_treatment_tiquant</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># download.file(</span>
<span class="co">#   url = paste0("http://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?save=efetch&amp;db=sra&amp;rettype=runinfo&amp;term=",</span>
<span class="co">#                "SRP066571"),</span>
<span class="co">#   destfile = "SRP066571runtable.txt",quiet = TRUE)</span>



<span class="co">### Load file with estimated cell fractions by quanTIseq:</span>
<span class="va">cdata</span> <span class="op">&lt;-</span> <span class="co"># read.csv(paste0(path, "/SRP066571_cell_fractions.txt"), header = TRUE, sep = "\t", stringsAsFactors = FALSE, row.names = 1)</span>
  <span class="va">ti_quant</span>


<span class="va">Song2017_MAPKi_treatment_tiquant</span>

<span class="va">ti_quant</span> <span class="op">&lt;-</span> <span class="fu">quantiseqr</span><span class="fu">::</span><span class="fu"><a href="../reference/extract_ti_from_se.html">extract_ti_from_se</a></span><span class="op">(</span><span class="va">Song2017_MAPKi_treatment_tiquant</span><span class="op">)</span>



<span class="co">## this will be adjusted!</span>


<span class="co">### Annotation</span>
<span class="va">gds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GEOquery/man/getGEO.html" class="external-link">getGEO</a></span><span class="op">(</span><span class="st">"GSE75299"</span><span class="op">)</span>
<span class="va">pdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">gds</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>

<span class="va">sraInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"SRP066571runtable.txt"</span>, package <span class="op">=</span> <span class="st">"quantiseqr"</span><span class="op">)</span>,
  header <span class="op">=</span> <span class="cn">TRUE</span>, sep <span class="op">=</span> <span class="st">","</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>

<span class="va">cdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">cdata</span>, <span class="va">sraInfo</span><span class="op">[</span>, <span class="st">"SampleName"</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span>
<span class="op">(</span><span class="va">cdata</span><span class="op">)</span>, <span class="va">sraInfo</span><span class="op">$</span><span class="va">Run</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">cdata</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cdata</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"geo_accession"</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">pdata</span><span class="op">)</span>
<span class="va">pdata</span> <span class="op">&lt;-</span> <span class="va">pdata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^Pt"</span>, <span class="va">pdata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="op">]</span>

<span class="va">pt_matchedinfo</span> <span class="op">&lt;-</span> <span class="va">pdata</span><span class="op">[</span>, <span class="st">"title"</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">cdata</span><span class="op">$</span><span class="va">geo_accession</span>, <span class="va">pdata</span><span class="op">$</span><span class="va">geo_accession</span><span class="op">)</span><span class="op">]</span>
<span class="va">cdata</span><span class="op">$</span><span class="va">pt_info</span> <span class="op">&lt;-</span> <span class="va">pt_matchedinfo</span>
<span class="co"># rownames(cdata) &lt;- cdata[, length(cdata)]</span>

<span class="va">cdata</span><span class="op">$</span><span class="va">groups</span> <span class="op">&lt;-</span> <span class="st">"POST"</span>
<span class="va">cdata</span><span class="op">$</span><span class="va">groups</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"baseline"</span>, <span class="va">cdata</span><span class="op">$</span><span class="va">pt_info</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"PRE"</span>

<span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">cdata</span><span class="op">)</span>

<span class="va">cdata</span>

<span class="va">cellfracs</span> <span class="op">&lt;-</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="va">cdata</span>, cols <span class="op">=</span> <span class="va">B.cells</span><span class="op">:</span><span class="va">Other</span><span class="op">)</span>

<span class="va">cellfracs</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">cellfracs</span><span class="op">$</span><span class="va">name</span>,
  levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
    <span class="st">"B.cells"</span>, <span class="st">"Macrophages.M1"</span>, <span class="st">"Macrophages.M2"</span>,
    <span class="st">"Monocytes"</span>, <span class="st">"Neutrophils"</span>, <span class="st">"NK.cells"</span>,
    <span class="st">"T.cells.CD4"</span>, <span class="st">"T.cells.CD8"</span>, <span class="st">"Tregs"</span>,
    <span class="st">"Dendritic.cells"</span>, <span class="st">"Other"</span>
  <span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">cellfracs</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">name</span>, y <span class="op">=</span> <span class="va">value</span>, x <span class="op">=</span> <span class="va">Sample</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>position <span class="op">=</span> <span class="st">"fill"</span>, stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"PuOr"</span><span class="op">)</span> <span class="op">+</span>
  <span class="co"># viridis::scale_fill_viridis(discrete = TRUE) +</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Cell Fractions"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>


<span class="va">prepost_data</span> <span class="op">&lt;-</span> <span class="va">cdata</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">cdata</span><span class="op">$</span><span class="va">pt_info</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">prepost_data</span> <span class="op">&lt;-</span> <span class="va">prepost_data</span><span class="op">[</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"Other"</span>, <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">prepost_data</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>

<span class="co">### Remove other cells and the last two columns</span>

<span class="co"># cdata &lt;- cdata[, -grep("Other", colnames(cdata))]</span>

<span class="co"># cdata &lt;- cdata[, -grep("geo_accession", colnames(cdata))]</span>

<span class="va">prepost_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="va">prepost_data</span>, cols <span class="op">=</span> <span class="va">B.cells</span><span class="op">:</span><span class="va">Dendritic.cells</span><span class="op">)</span>

<span class="va">prepost_data</span><span class="op">$</span><span class="va">groups</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">prepost_data</span><span class="op">$</span><span class="va">groups</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PRE"</span>, <span class="st">"POST"</span><span class="op">)</span><span class="op">)</span>

<span class="va">prepost_data</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">prepost_data</span><span class="op">$</span><span class="va">name</span>,
  levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>
    <span class="st">"B.cells"</span>, <span class="st">"Macrophages.M1"</span>, <span class="st">"Macrophages.M2"</span>,
    <span class="st">"Monocytes"</span>, <span class="st">"Neutrophils"</span>, <span class="st">"NK.cells"</span>,
    <span class="st">"T.cells.CD4"</span>, <span class="st">"T.cells.CD8"</span>, <span class="st">"Tregs"</span>,
    <span class="st">"Dendritic.cells"</span>
  <span class="op">)</span>
<span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">prepost_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">name</span>, <span class="va">value</span>, fill <span class="op">=</span> <span class="va">groups</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"cell fractions"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>
<span class="co"># +</span>
<span class="co">#   stat_compare_means(aes(group = groups),</span>
<span class="co">#     symnum.args = list(cutpoints = c(0, 0.001, 0.01, 0.05, 0.1, 1), symbols = c("****", "***", "**", "*", "ns")),</span>
<span class="co">#     label = "p.signif",</span>
<span class="co">#     hide.ns = TRUE, cex = 10</span>
<span class="co">#   ) # default: Wilcoxon Test</span></code></pre></div>
</div>
<div id="use-case-4-running-on-simulated-data-for-validation" class="section level2">
<h2 class="hasAnchor">
<a href="#use-case-4-running-on-simulated-data-for-validation" class="anchor" aria-hidden="true"></a>Use case 4: Running on simulated data for validation</h2>
<p>We will now display the performance of <code>quantiseqr</code> on a dataset for which the ground truth parameters are known.<br>
The large simulated dataset represents the RNA-seq expression data from breast tumors with different immune-infiltration scenarios, consisting of a total of 1700 samples, that were generated by mixing RNA-seq reads from purified immune cell types and from a MCF7 breast tumor cell line.</p>
<p>These samples were generated considering different immune relative cell proportions, tumor purity values (0:10:100%), and at different sequencing depths (1, 2, 5, 10, 20, 50, and 100 million read pairs).</p>
<p>Please refer to <a href="https://icbi.i-med.ac.at/software/quantiseq/doc/" class="external-link uri">https://icbi.i-med.ac.at/software/quantiseq/doc/</a> and to <span class="citation">(Finotello et al. 2019)</span> for more details.</p>
<p>This dataset is coupled with a table where the original information on the used true fractions is available, and can be used to benchmark the performance of the deconvolution algorithm.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># downloading first the file from https://icbi.i-med.ac.at/software/quantiseq/doc/downloads/quanTIseq_SimRNAseq_mixture.txt</span>

<span class="co"># https://icbi.i-med.ac.at/software/quantiseq/doc/</span>

<span class="va">tpm_1700mixtures</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html" class="external-link">read_tsv</a></span><span class="op">(</span><span class="st">"quanTIseq_SimRNAseq_mixture.txt.gz"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">tpm_1700mixtures</span><span class="op">)</span>

<span class="co"># extracting the gene names, restructuring the matrix by dropping the column</span>
<span class="va">tpm_genesymbols</span> <span class="op">&lt;-</span> <span class="va">tpm_1700mixtures</span><span class="op">$</span><span class="va">Gene</span>
<span class="va">tpm_1700mixtures</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">tpm_1700mixtures</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tpm_1700mixtures</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tpm_genesymbols</span>

<span class="co"># running quantiseq on that set</span>
<span class="co"># True mRNA fractions were simulated with no total-mRNA bias. Thus, these data should be analyzed specifying the option scale_mRNA set to FALSE</span>
<span class="va">ti_quant_sim1700mixtures</span> <span class="op">&lt;-</span> <span class="fu">quantiseqr</span><span class="fu">::</span><span class="fu"><a href="../reference/run_quantiseq.html">run_quantiseq</a></span><span class="op">(</span>
  expression_data <span class="op">=</span> <span class="va">tpm_1700mixtures</span>,
  signature_matrix <span class="op">=</span> <span class="st">"TIL10"</span>,
  is_arraydata <span class="op">=</span> <span class="cn">FALSE</span>,
  is_tumordata <span class="op">=</span> <span class="cn">TRUE</span>,
  scale_mRNA <span class="op">=</span> <span class="cn">FALSE</span>
<span class="op">)</span>

<span class="co"># save(ti_quant_sim1700mixtures, file = "data/ti_quant_sim1700mixtures.RData")</span></code></pre></div>
<p>To avoid the download of a large file, we provide the precomputed object <code>ti_quant_sim1700mixtures</code> in the <code>quantiseqr</code> package - the chunk above is still fully functional once the mixture file has been retrieved.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ti_quant_sim1700mixtures</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">ti_quant_sim1700mixtures</span><span class="op">)</span>
<span class="co">#&gt; [1] 1700   12</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ti_quant_sim1700mixtures</span><span class="op">)</span>
<span class="co">#&gt;                  Sample     B.cells Macrophages.M1 Macrophages.M2  Monocytes</span>
<span class="co">#&gt; SimData1       SimData1 0.019844435   0.0597686387     0.27196959 0.23806843</span>
<span class="co">#&gt; SimData10     SimData10 0.009382905   0.0249402719     0.08979368 0.09425353</span>
<span class="co">#&gt; SimData100   SimData100 0.033495650   0.0096198517     0.03057850 0.00000000</span>
<span class="co">#&gt; SimData1000 SimData1000 0.042487150   0.0149141815     0.05204647 0.03813509</span>
<span class="co">#&gt; SimData1001 SimData1001 0.029230069   0.0139994722     0.02193008 0.01626132</span>
<span class="co">#&gt; SimData1002 SimData1002 0.013160627   0.0003221679     0.02475316 0.03709523</span>
<span class="co">#&gt;             Neutrophils   NK.cells T.cells.CD4 T.cells.CD8       Tregs</span>
<span class="co">#&gt; SimData1    0.002813382 0.10095469  0.00000000  0.08827795 0.169366189</span>
<span class="co">#&gt; SimData10   0.005454131 0.03922658  0.00000000  0.04532855 0.067272535</span>
<span class="co">#&gt; SimData100  0.016667177 0.04231232  0.00000000  0.04674569 0.010138803</span>
<span class="co">#&gt; SimData1000 0.032586376 0.03811065  0.03650752  0.03029464 0.017042234</span>
<span class="co">#&gt; SimData1001 0.021034018 0.02682053  0.02145636  0.01624266 0.011879932</span>
<span class="co">#&gt; SimData1002 0.007311373 0.01710432  0.00000000  0.01314424 0.008695806</span>
<span class="co">#&gt;             Dendritic.cells     Other</span>
<span class="co">#&gt; SimData1        0.048936698 0.0000000</span>
<span class="co">#&gt; SimData10       0.015406051 0.6089418</span>
<span class="co">#&gt; SimData100      0.023892748 0.7865493</span>
<span class="co">#&gt; SimData1000     0.015217518 0.6826582</span>
<span class="co">#&gt; SimData1001     0.028108596 0.7930370</span>
<span class="co">#&gt; SimData1002     0.001104581 0.8773085</span>
<span class="fu"><a href="../reference/quantiplot.html">quantiplot</a></span><span class="op">(</span><span class="va">ti_quant_sim1700mixtures</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span>, <span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="using_quantiseqr_files/figure-html/ex4-load-plot-1.png" width="576"></p>
<p>We also read in the true proportions, known by design - this is provided as a text file inside <code>quantiseqr</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">true_prop_1700mix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span>
  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"quanTIseq_SimRNAseq_read_fractions.txt.gz"</span>, package <span class="op">=</span> <span class="st">"quantiseqr"</span><span class="op">)</span>,
  sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">true_prop_1700mix</span><span class="op">)</span>
<span class="co">#&gt;           Mix    B.cells Macrophages.M1 Macrophages.M2  Monocytes Neutrophils</span>
<span class="co">#&gt; 1    SimData1 0.01890000    0.067800000     0.22910000 0.22560000 0.007200000</span>
<span class="co">#&gt; 2   SimData10 0.00756000    0.027120000     0.09164000 0.09024000 0.002880000</span>
<span class="co">#&gt; 3  SimData100 0.03178064    0.017140343     0.01076022 0.02004040 0.019540391</span>
<span class="co">#&gt; 4 SimData1000 0.04103877    0.009149726     0.04553863 0.05792826 0.028139156</span>
<span class="co">#&gt; 5 SimData1001 0.02735945    0.006099878     0.03035939 0.03861923 0.018759625</span>
<span class="co">#&gt; 6 SimData1002 0.01367986    0.003049970     0.01517985 0.01930981 0.009379906</span>
<span class="co">#&gt;     NK.cells T.cells.CD4 T.cells.CD8       Tregs Dendritic.cells    Tumor</span>
<span class="co">#&gt; 1 0.09700000  0.00960000 0.108700000 0.203900000     0.032200000 0.000000</span>
<span class="co">#&gt; 2 0.03880000  0.00384000 0.043480000 0.081560000     0.012880000 0.600000</span>
<span class="co">#&gt; 3 0.03480070  0.02140043 0.036440729 0.001720034     0.006360127 0.800016</span>
<span class="co">#&gt; 4 0.03437897  0.03677890 0.029279122 0.012719618     0.005069848 0.699979</span>
<span class="co">#&gt; 5 0.02291954  0.02451951 0.019519610 0.008479830     0.003379932 0.799984</span>
<span class="co">#&gt; 6 0.01145989  0.01225988 0.009759902 0.004239958     0.001689983 0.899991</span></code></pre></div>
<p>In the following chunk we perform some preprocessing steps to facilitate the comparison, also in a graphical manner.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># merging the two sets to facilitate the visualization</span>
<span class="co"># colnames(ti_quant_sim1700mixtures) &lt;- paste0("quantiseq_", colnames(ti_quant_sim1700mixtures))</span>
<span class="co"># colnames(true_prop_1700mix) &lt;- paste0("trueprops_", colnames(true_prop_1700mix))</span>

<span class="co"># ti_quant_sim1700mixtures$method &lt;- "quanTIseq"</span>
<span class="co"># true_prop_1700mix$method &lt;- "ground_truth"</span>

<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">true_prop_1700mix</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Sample"</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">true_prop_1700mix</span><span class="op">)</span><span class="op">[</span><span class="fl">12</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Other"</span>

<span class="va">ti_long</span> <span class="op">&lt;-</span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="va">ti_quant_sim1700mixtures</span>,
  cols <span class="op">=</span> <span class="va">B.cells</span><span class="op">:</span><span class="va">Other</span>,
  names_to <span class="op">=</span> <span class="st">"cell_type"</span>,
  values_to <span class="op">=</span> <span class="st">"value_quantiseq"</span>
<span class="op">)</span>
<span class="va">ti_long</span><span class="op">$</span><span class="va">mix_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">ti_long</span><span class="op">$</span><span class="va">Sample</span>, <span class="va">ti_long</span><span class="op">$</span><span class="va">cell_type</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>

<span class="va">tp_long</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="va">true_prop_1700mix</span>,
  cols <span class="op">=</span> <span class="va">B.cells</span><span class="op">:</span><span class="va">Other</span>,
  names_to <span class="op">=</span> <span class="st">"cell_type"</span>,
  values_to <span class="op">=</span> <span class="st">"value_trueprop"</span>
<span class="op">)</span>
<span class="va">tp_long</span><span class="op">$</span><span class="va">mix_id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">tp_long</span><span class="op">$</span><span class="va">Sample</span>, <span class="va">tp_long</span><span class="op">$</span><span class="va">cell_type</span>, sep <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span>


<span class="va">ti_tp_merged</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">ti_long</span>, <span class="va">tp_long</span>, by <span class="op">=</span> <span class="st">"mix_id"</span><span class="op">)</span>
<span class="va">ti_tp_merged</span><span class="op">$</span><span class="va">cell_type.x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">ti_tp_merged</span><span class="op">$</span><span class="va">cell_type.x</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">true_prop_1700mix</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">12</span><span class="op">]</span><span class="op">)</span>

<span class="co"># ti_merged &lt;- rbind(ti_quant_sim1700mixtures,</span>
<span class="co"># true_prop_1700mix)</span>

<span class="co"># ti_merged_long &lt;- pivot_longer(ti_quant_sim1700mixtures, cols = B.cells:Other)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>
  <span class="va">ti_tp_merged</span>,
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
    x <span class="op">=</span> <span class="va">value_trueprop</span>,
    y <span class="op">=</span> <span class="va">value_quantiseq</span>,
    col <span class="op">=</span> <span class="va">cell_type.x</span>
  <span class="op">)</span>
<span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="using_quantiseqr_files/figure-html/ex4-compare-plot-1.png" width="672"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span>
  <span class="va">ti_tp_merged</span>,
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>
    x <span class="op">=</span> <span class="va">value_trueprop</span>,
    y <span class="op">=</span> <span class="va">value_quantiseq</span>,
    col <span class="op">=</span> <span class="va">cell_type.x</span>
  <span class="op">)</span>
<span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">cell_type.x</span>, scales <span class="op">=</span> <span class="st">"free"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="using_quantiseqr_files/figure-html/ex4-compare-plot-2.png" width="672"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># for (i in colnames(ti_quant_sim1700mixtures)) {</span>
<span class="co">#   message(i)</span>
<span class="co">#   if(i!="Sample")</span>
<span class="co">#     plot(true_prop_1700mix[[i]], ti_quant_sim1700mixtures[[i]], main = i)</span>
<span class="co"># }</span>
<span class="co"># plot( true_prop_1700mix$Tumor, ti_quant_sim1700mixtures$Other, main = "other")</span></code></pre></div>
<p>This figure aims to replicate with live code the one available as <a href="https://static-content.springer.com/esm/art%3A10.1186%2Fs13073-019-0638-6/MediaObjects/13073_2019_638_MOESM2_ESM.pdf" class="external-link">Supplementary Figure 1</a>.</p>
<p>…</p>
</div>
</div>
<div id="faqs" class="section level1">
<h1 class="hasAnchor">
<a href="#faqs" class="anchor" aria-hidden="true"></a>FAQs</h1>
<p><strong>Q: Do I have to provide my expression data formatted as TPMs? Why is that so?</strong></p>
<p>A: The expression data is indeed expected to be provided as TPM values. <code>quantiseqr</code> might warn you if you are providing a different format (counts, normalized counts) - this does not mean that it will trigger an error as the computation is still able to proceed.<br>
Still: it is not the recommended way. If using a <code>SummarizedExperiment</code> object coming from Salmon’s quantifications, the <code>tximeta</code>/<code>tximport</code> pipeline will provide an assay named “abundance”, which would be handled internally by the <code><a href="../reference/se_to_matrix.html">se_to_matrix()</a></code> function - you can simply call <code>quantiseqr()</code> and provide the <code>SummarizedExperiment</code> object as main parameter.</p>
<p><strong>Q: Can I use <code>quantiseqr</code> with samples from model systems, i.e. not from human?</strong></p>
<p>A: You can exploit orthology-based conversions among gene identifiers to use <code>quantiseqr</code> e.g. in mouse scenarios. Keep in mind, though, that the TIL10 signature has been explicitly designed and validated on human samples.</p>
<p><strong>Q: My expression data is encoding the features in a different identifier than Gene Symbols. Can I use <code>quantiseqr</code> for that?</strong></p>
<p>A: Sure, just make sure to convert the identifiers beforehand - you can use one of the many options available inside Bioconductor for streamlining this step (e.g. the <code>org.Hs.eg.db</code> and the function <code><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">AnnotationDbi::mapIds()</a></code>).</p>
<p><strong>Q: I’m interested in other such deconvolution methods. What other options are available?</strong></p>
<p>A: You can check out the works of <span class="citation">(Sturm et al. 2019, @Sturm2020)</span> to find a collection of methods, provided in the <code>immunedeconv</code> package, and benchmarked in the above mentioned manuscripts.</p>
<p><strong>Q: Can I provide my own signature like the <code>TIL10</code> and use that in <code>quantiseqr</code>?</strong></p>
<p>A: Yes, we recommend you to read up on the methods used in <span class="citation">(Finotello et al. 2019)</span> to see what processing steps have been applied, and additionally have a look at the formatting requirements by inspecting the <code>TIL10_*</code> files provided in the <code>extdata</code> folder of the installed package.</p>
</div>
<div id="session-info" class="section level1 unnumbered smaller">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor" aria-hidden="true"></a>Session Info</h1>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R Under development (unstable) (2021-03-29 r80130)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Catalina 10.15.7</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] reshape2_1.4.4      GEOquery_2.59.0     Biobase_2.51.0     </span>
<span class="co">#&gt; [4] BiocGenerics_0.37.2 tibble_3.1.1        tidyr_1.1.3        </span>
<span class="co">#&gt; [7] ggplot2_3.3.3       dplyr_1.0.5         quantiseqr_0.99.1  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] MatrixGenerics_1.3.1        sass_0.3.1                 </span>
<span class="co">#&gt;  [3] jsonlite_1.7.2              bslib_0.2.4                </span>
<span class="co">#&gt;  [5] highr_0.9                   BiocManager_1.30.12        </span>
<span class="co">#&gt;  [7] stats4_4.1.0                GenomeInfoDbData_1.2.4     </span>
<span class="co">#&gt;  [9] yaml_2.2.1                  pillar_1.6.0               </span>
<span class="co">#&gt; [11] lattice_0.20-41             glue_1.4.2                 </span>
<span class="co">#&gt; [13] limma_3.47.12               quadprog_1.5-8             </span>
<span class="co">#&gt; [15] digest_0.6.27               RColorBrewer_1.1-2         </span>
<span class="co">#&gt; [17] GenomicRanges_1.43.4        XVector_0.31.1             </span>
<span class="co">#&gt; [19] colorspace_2.0-0            plyr_1.8.6                 </span>
<span class="co">#&gt; [21] htmltools_0.5.1.1           preprocessCore_1.53.2      </span>
<span class="co">#&gt; [23] Matrix_1.3-2                lpSolve_5.6.15             </span>
<span class="co">#&gt; [25] pkgconfig_2.0.3             zlibbioc_1.37.0            </span>
<span class="co">#&gt; [27] purrr_0.3.4                 scales_1.1.1               </span>
<span class="co">#&gt; [29] farver_2.1.0                generics_0.1.0             </span>
<span class="co">#&gt; [31] IRanges_2.25.9              ellipsis_0.3.1             </span>
<span class="co">#&gt; [33] cachem_1.0.4                withr_2.4.2                </span>
<span class="co">#&gt; [35] SummarizedExperiment_1.21.3 cli_2.4.0                  </span>
<span class="co">#&gt; [37] magrittr_2.0.1              crayon_1.4.1               </span>
<span class="co">#&gt; [39] ps_1.6.0                    memoise_2.0.0              </span>
<span class="co">#&gt; [41] evaluate_0.14               fs_1.5.0                   </span>
<span class="co">#&gt; [43] fansi_0.4.2                 MASS_7.3-53.1              </span>
<span class="co">#&gt; [45] xml2_1.3.2                  textshaping_0.3.3          </span>
<span class="co">#&gt; [47] tools_4.1.0                 hms_1.0.0                  </span>
<span class="co">#&gt; [49] BiocStyle_2.19.2            lifecycle_1.0.0            </span>
<span class="co">#&gt; [51] matrixStats_0.58.0          stringr_1.4.0              </span>
<span class="co">#&gt; [53] S4Vectors_0.29.15           munsell_0.5.0              </span>
<span class="co">#&gt; [55] DelayedArray_0.17.10        limSolve_1.5.6             </span>
<span class="co">#&gt; [57] compiler_4.1.0              pkgdown_1.6.1.9001         </span>
<span class="co">#&gt; [59] jquerylib_0.1.3             GenomeInfoDb_1.27.11       </span>
<span class="co">#&gt; [61] systemfonts_1.0.1           rlang_0.4.10               </span>
<span class="co">#&gt; [63] grid_4.1.0                  RCurl_1.98-1.3             </span>
<span class="co">#&gt; [65] rstudioapi_0.13             labeling_0.4.2             </span>
<span class="co">#&gt; [67] bitops_1.0-6                rmarkdown_2.7              </span>
<span class="co">#&gt; [69] gtable_0.3.0                curl_4.3                   </span>
<span class="co">#&gt; [71] DBI_1.1.1                   R6_2.5.0                   </span>
<span class="co">#&gt; [73] knitr_1.32                  fastmap_1.1.0              </span>
<span class="co">#&gt; [75] utf8_1.2.1                  rprojroot_2.0.2            </span>
<span class="co">#&gt; [77] ragg_1.1.2                  readr_1.4.0                </span>
<span class="co">#&gt; [79] desc_1.3.0                  stringi_1.5.3              </span>
<span class="co">#&gt; [81] Rcpp_1.0.6                  vctrs_0.3.7                </span>
<span class="co">#&gt; [83] tidyselect_1.1.0            xfun_0.22</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor" aria-hidden="true"></a>References</h1>
<div id="refs" class="references">
<div id="ref-quantiseq2019">
<p>Finotello, Francesca, Clemens Mayer, Christina Plattner, Gerhard Laschober, Dietmar Rieder, Hubert Hackl, Anne Krogsdam, et al. 2019. “Molecular and Pharmacological Modulators of the Tumor Immune Contexture Revealed by Deconvolution of Rna-Seq Data.” <em>Genome Medicine</em> 11 (1): 34.</p>
</div>
<div id="ref-Plattner2020">
<p>Plattner, Christina, Francesca Finotello, and Dietmar Rieder. 2020. <em>Deconvoluting tumor-infiltrating immune cells from RNA-seq data using quanTIseq</em>. 1st ed. Vol. 636. Elsevier Inc. <a href="https://doi.org/10.1016/bs.mie.2019.05.056" class="external-link">https://doi.org/10.1016/bs.mie.2019.05.056</a>.</p>
</div>
<div id="ref-EPIC2017">
<p>Racle, Julien, Kaat de Jonge, Petra Baumgaertner, Daniel E Speiser, and David Gfeller. 2017. “Simultaneous enumeration of cancer and immune cell types from bulk tumor gene expression data.” <em>eLife</em> 6 (November): e26476. <a href="https://doi.org/10.7554/eLife.26476" class="external-link">https://doi.org/10.7554/eLife.26476</a>.</p>
</div>
<div id="ref-Song2017">
<p>Song, Chunying, Marco Piva, Lu Sun, Aayoung Hong, Gatien Moriceau, Xiangju Kong, Hong Zhang, et al. 2017. “Recurrent Tumor CellIntrinsic and Extrinsic Alterations During MAPKi-Induced Melanoma Regression and Early Adaptation.” <em>Cancer Discovery</em> 7 (11): 1248–65. <a href="https://doi.org/10.1158/2159-8290.cd-17-0401" class="external-link">https://doi.org/10.1158/2159-8290.cd-17-0401</a>.</p>
</div>
<div id="ref-Sturm2020">
<p>Sturm, Gregor, Francesca Finotello, and Markus List. 2020. “Immunedeconv: An R Package for Unified Access to Computational Methods for Estimating Immune Cell Fractions from Bulk RNA-Sequencing Data.” In, 2120:223–32. Berlin, Boston: DE GRUYTER. <a href="https://doi.org/10.1007/978-1-0716-0327-7_16" class="external-link">https://doi.org/10.1007/978-1-0716-0327-7_16</a>.</p>
</div>
<div id="ref-sturm2019">
<p>Sturm, Gregor, Francesca Finotello, Florent Petitprez, Jitao David Zhang, Jan Baumbach, Wolf H Fridman, Markus List, and Tatsiana Aneichyk. 2019. “Comprehensive Evaluation of Transcriptome-Based Cell-Type Quantification Methods for Immuno-Oncology.” <em>Bioinformatics</em> 35 (14): i436–i445.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:marinif@uni-mainz.de" class="email">marinif@uni-mainz.de</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p><a href="mailto:francesca.finotello@i-med.ac.at" class="email">francesca.finotello@i-med.ac.at</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Federico Marini, Francesca Finotello.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link external-link">pkgdown</a> 1.6.1.9001.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
